File: CloudStationWeb/Models/FunVasosModels.cs
namespace CloudStationWeb.Models
{
    public class FunVasosHorario
    {
        public DateTime Ts { get; set; }
        public string Presa { get; set; } = string.Empty;
        public short Hora { get; set; }
        public float? Elevacion { get; set; }
        public float? Almacenamiento { get; set; }
        public float? Diferencia { get; set; }
        public float? AportacionesQ { get; set; }
        public float? AportacionesV { get; set; }
        public float? ExtraccionesTurbQ { get; set; }
        public float? ExtraccionesTurbV { get; set; }
        public float? ExtraccionesVertQ { get; set; }
        public float? ExtraccionesVertV { get; set; }
        public float? ExtraccionesTotalQ { get; set; }
        public float? ExtraccionesTotalV { get; set; }
        public float? Generacion { get; set; }
        public short? NumUnidades { get; set; }
        public float? AportacionCuencaPropia { get; set; }
        public float? AportacionPromedio { get; set; }
    }

    public class FunVasosResumenPresa
    {
        public string Presa { get; set; } = string.Empty;
        public float? UltimaElevacion { get; set; }
        public float? UltimoAlmacenamiento { get; set; }
        public float? TotalAportacionesV { get; set; }
        public float? TotalExtraccionesV { get; set; }
        public float? TotalGeneracion { get; set; }
        public short UltimaHora { get; set; }
        public List<FunVasosHorario> Datos { get; set; } = new();
    }

    public class FunVasosViewModel
    {
        public DateTime FechaInicio { get; set; }
        public DateTime FechaFin { get; set; }
        public List<FunVasosResumenPresa> Presas { get; set; } = new();
        public List<DateTime> FechasDisponibles { get; set; } = new();
    }
}

File: ChatiOS/CloudStationChat/Models/FunVasosModels.swift
import Foundation

// MARK: - API Response Models

struct FunVasosViewModel: Codable {
    let fechaInicio: String
    let fechaFin: String
    let presas: [FunVasosResumenPresa]
    let fechasDisponibles: [String]
}

struct FunVasosResumenPresa: Codable, Identifiable {
    var id: String { presa }
    let presa: String
    let ultimaElevacion: Float?
    let ultimoAlmacenamiento: Float?
    let totalAportacionesV: Float?
    let totalExtraccionesV: Float?
    let totalGeneracion: Float?
    let ultimaHora: Int
    let datos: [FunVasosHorario]
}

struct FunVasosHorario: Codable, Identifiable {
    var id: String { "\(ts)-\(presa)-\(hora)" }
    let ts: String
    let presa: String
    let hora: Int
    let elevacion: Float?
    let almacenamiento: Float?
    let diferencia: Float?
    let aportacionesQ: Float?
    let aportacionesV: Float?
    let extraccionesTurbQ: Float?
    let extraccionesTurbV: Float?
    let extraccionesVertQ: Float?
    let extraccionesVertV: Float?
    let extraccionesTotalQ: Float?
    let extraccionesTotalV: Float?
    let generacion: Float?
    let numUnidades: Int?
    let aportacionCuencaPropia: Float?
    let aportacionPromedio: Float?
}

struct CascadeResponse: Codable {
    let presas: [CascadePresa]
    let fecha: String
}

struct CascadePresa: Codable, Identifiable {
    var id: String { key }
    let key: String
    let name: String
    let currentElev: Float?
    let generation: Float?
    let activeUnits: Int?
    let almacenamiento: Float?
    let ultimaHora: Int?
    let fecha: String?
    let aportacionesV: Float?
    let extraccionesV: Float?
}
